src/cdomains/stringDomain.ml

```diff
@@ -0,0 +1,97 @@
+include Printable.StdLeaf
+
+let name () = "string"
+
+type t = string option [@@deriving eq, ord, hash]
+
+let hash x =
+  if GobConfig.get_string "ana.base.strings.domain" = "disjoint" then
```

Comment:
It may make sense to cache the value for `"ana.base.strings.domain"` in some Variant typed `ref`.

Reply:
I now added a `ref` caching the value of this configuration.